homeassistant/components/wilight/config_flow.py

```diff
@@ -0,0 +1,111 @@
+"""Config flow to configure WiLight."""
+import logging
+from urllib.parse import urlparse
+
+import pywilight
+
+from homeassistant.components import ssdp
+from homeassistant.config_entries import CONN_CLASS_LOCAL_PUSH, ConfigFlow
+from homeassistant.const import CONF_HOST
+
+from . import PLATFORMS
+from .const import DOMAIN  # pylint: disable=unused-import
+
+CONF_SERIAL_NUMBER = "serial_number"
+CONF_MODEL_NAME = "model_name"
+
+WILIGHT_MANUFACTURER = "All Automacao Ltda"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class WiLightFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a WiLight config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_LOCAL_PUSH
+
+    # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+    def __init__(self):
+        """Initialize the WiLight flow."""
+        self._host = None
+        self._serial_number = None
+        self._device_id = None
+        self._model_name = None
+        self._components = []
+        self._components_text = ""
+
+    def _self_update(self, host, serial_number, model_name):
+        self._host = host
+        self._serial_number = serial_number
+        self._device_id = f"WL{serial_number}"
+        self._model_name = model_name
+        self._components = pywilight.get_components_from_model(model_name)
```

Comment:
Does this do I/O?

Reply:
A WiLight device can have more than one component, for example light and fan.
In this first integration I am only including light, taking into account the guidelines for not having a very extensive initial integration.
pywilight.get_components_from_model (model_name)
It is a function that lists the components that that WiLight model has.

Reply:
So it won't do I/O?

Reply:
I checked the library. It's safe.